python-avd/pyavd/_eos_designs/structured_config/overlay/router_bgp.py

```diff
@@ -222,15 +225,21 @@ def _address_family_evpn(self: AvdStructuredConfigOverlay) -> dict | None:
         peer_groups = []
 
         overlay_peer_group = {}
-        if self.shared_utils.overlay_evpn_vxlan is True:
-            if self.shared_utils.is_wan_router:
-                overlay_peer_group = {
-                    "name": self.inputs.bgp_peer_groups.wan_overlay_peers.name,
-                    "activate": True,
-                    "encapsulation": self.inputs.wan_encapsulation,
-                }
-            else:
-                overlay_peer_group = {"name": self.inputs.bgp_peer_groups.evpn_overlay_peers.name, "activate": True}
+        if self.shared_utils.is_wan_router:
+            overlay_peer_group = {
+                "name": self.inputs.bgp_peer_groups.wan_overlay_peers.name,
+                "activate": True,
+                "encapsulation": self.inputs.wan_encapsulation,
+            }
+            if self.shared_utils.wan_role != "server":
+                overlay_peer_group.update(
+                    {
+                        "route_map_in": "RM-EVPN-SOO-IN",
+                        "route_map_out": "RM-EVPN-SOO-OUT",
+                    },
+                )
+        elif self.shared_utils.overlay_evpn_vxlan is True:
```

Comment:
I think this should be a regular if.

Reply:
done

Reply:
actually this now add a group that was not there before.

Reply:
solved with the other change to evpn_role